test/stats_test.go

```diff
@@ -32,91 +31,116 @@ import (
 	"google.golang.org/grpc/stats"
 )
 
-// TestPeerForClientStatsHandler tests the scenario where stats handler
-// (having peer as part of its struct) has peer enriched as part of
-// stream context.
+// TestPeerForClientStatsHandler configures a stats handler that
+// verifies that peer is sent for OutPayload, InPayload, End
+// stats handlers.
 func (s) TestPeerForClientStatsHandler(t *testing.T) {
-	spy := &handlerSpy{}
+	statsHandler := &peerStatsHandler{}
+
+	// Define expected stats callouts and whether a peer object should be populated.
+	// Note:
+	//   * Begin stats don't have peer information as the RPC begins before peer resolution.
+	//   * PickerUpdated stats don't have peer information as the picker operates without transport-level knowledge.
```

Comment:
Please wrap comments to 80 characters. The guidance for this repo is comments wrapped to 80 characters, but we do not have a length maximum for code line characters.

Reply:
done